website/addons/osfstorage/views.py

```diff
@@ -217,6 +222,10 @@ def osfstorage_delete(file_node, payload, node_addon, **kwargs):
 
     except exceptions.FileNodeCheckedOutError:
         raise HTTPError(httplib.FORBIDDEN)
+    except exceptions.FileNodeIsPrimaryFile:
+        raise HTTPError(httplib.FORBIDDEN, data={
```

Comment:
@brianjgeiger, making those changes. However, that is the behavior used for checked-out files in both cases (move and delete).